lib/down/net_http.rb

```diff
@@ -158,7 +158,8 @@ def open_uri(uri, options, follows_remaining:)
 
       # forward cookies on the redirect
       if !exception.io.meta["set-cookie"].to_s.empty?
-        options["Cookie"] = exception.io.meta["set-cookie"]
+        options["Cookie"] ||= ''        
+        options["Cookie"] += "#{exception.io.meta["set-cookie"]}"
```

Comment:
I did a change here. I was retrieving only the first cookie (exception.io.meta["set-cookie"].split(';')[0].to_s) and is ok as it is.

The only change is concatenate to options['Cookie'] the new cookies received.

Reply:
Thanks for note! Would you have to de-dup before just concatenating? Am I wrong that many servers will send the same cookies again repeatedly, and concatenating will wind up with multiple copies of each cookie and potentially an unnecessarily large cookie header? 

My impression was the same as I think janko's, that the behavior you are trying to fix is unusual, but the usual behavior is to re-send the same cookies on every request. 

Also, are we worried about cross-site restrictions here, if the redirect crosses site boundaries?

Reply:
> Am I wrong that many servers will send the same cookies again repeatedly, and concatenating will wind up with multiple copies of each cookie and potentially an unnecessarily large cookie header?

Yes, I agree. This is a simple approach because the cookie is going to be detected as duplicated in the server if the name, path and domain are the same and it's going to use the last one. 

I can work on another approach to avoid duplication.

> My impression was the same as I think janko's, that the behavior you are trying to fix is unusual, but the usual behavior is to re-send the same cookies on every request.

Agree, although unusual, sometimes occurs and would be great to have a solution.

> Also, are we worried about cross-site restrictions here, if the redirect crosses site boundaries?

I don't understand this. Could you explain more deeply? We are following the redirections from the server and we set the cookie that the server returns. If the cross-site boundaries are surpassed would be an error of the server who redirect you and returned the set-cookie header.